dotCMS/src/main/java/com/dotcms/security/secret/SecretsStoreKeyStoreImpl.java

```diff
@@ -0,0 +1,352 @@
+            putInCache(variableKey, encryptedVal);
+        } catch (Exception e) {
+            Logger.warn(this.getClass(), "Unable to save secret from " + SECRETS_STORE_FILE + ": " + e);
+            throw new DotRuntimeException(e);
+        }
+        return true;
+    }
+
+    /**
+     * deletes a value from the store
+     */
+    @Override
+    public void deleteValue(final String secretKey) {
+        try {
+            final KeyStore keyStore = getSecretsStore();
+            keyStore.deleteEntry(secretKey);
+            saveSecretsStore(keyStore);
+            flushCache(secretKey);
+        } catch (KeyStoreException | DotCacheException e) {
+            Logger.warn(this.getClass(), "Unable to delete secret from  " + SECRETS_STORE_FILE + ": " + e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private Key key() {
+        return Sneaky.sneak(() -> APILocator.getCompanyAPI().getDefaultCompany()).getKeyObj();
+    }
+
+    @VisibleForTesting
+    protected String encrypt(final char[] val) {
+        return encrypt(new String(val));
+    }
+
+    @VisibleForTesting
+    protected String encrypt(final String val) {
+        return Sneaky.sneak(() -> Encryptor.encrypt(key(), val));
+    }
+
+    @VisibleForTesting
+    protected String digest(final String val) {
+        return Sneaky.sneak(() -> Encryptor.digest(val));
+    }
+
+    @VisibleForTesting
+    protected char[] decrypt(final String encryptedString) {
+        if (encryptedString == null || encryptedString.length() == 0) {
+            return null;
+        }
+        return Sneaky.sneak(() -> Encryptor.decrypt(key(), encryptedString).toCharArray());
+    }
+
+    @VisibleForTesting
+    protected char[] decrypt(final char[] encryptedString) {
+        if (encryptedString == null || encryptedString.length == 0) {
+            return null;
+        }
+        return decrypt(new String(encryptedString));
+    }
+
+    private char[] getFromCache(final String key, final Supplier<char[]> defaultValue) {
+        char[] retVal = (char[]) CacheLocator.getCacheAdministrator().getNoThrow(key, SECRETS_CACHE_GROUP);
+        if (retVal == null) {
+            retVal = defaultValue.get();
+            CacheLocator.getCacheAdministrator().put(key, retVal, SECRETS_CACHE_GROUP);
+        }
+        return retVal;
+    }
+
+    private synchronized void putInCache(final String key, final char[] val)
+            throws KeyStoreException, DotCacheException {
+        CacheLocator.getCacheAdministrator().put(key, val ,SECRETS_CACHE_GROUP);
+        putInCache(key);
+    }
+
+    private synchronized void putInCache(final String key) throws KeyStoreException, DotCacheException{
+         final Set <String> keys =  getKeysFromCache();
+         keys.add(key);
+         CacheLocator.getCacheAdministrator().put(SECRETS_CACHE_KEY, keys, SECRETS_CACHE_KEYS_GROUP);
+    }
+
+   private synchronized Set<String> getKeysFromCache() throws KeyStoreException, DotCacheException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use block level rather than method level synchronization](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)